connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java

```diff
@@ -838,39 +733,32 @@ public void testPutConnectorConfig() throws Exception {
         newConnConfig.put("foo", "bar");
 
-        Callback<Map<String, String>> connectorConfigCb = PowerMock.createMock(Callback.class);
-        // Callback<Herder.Created<ConnectorInfo>> putConnectorConfigCb = PowerMock.createMock(Callback.class);
+        Callback<Map<String, String>> connectorConfigCb = mock(Callback.class);
 
         // Create
-        connector = PowerMock.createMock(BogusSourceConnector.class);
+        connector = mock(BogusSourceConnector.class);
         expectAdd(SourceSink.SOURCE);
-        Connector connectorMock = PowerMock.createMock(SourceConnector.class);
+        Connector connectorMock = mock(SourceConnector.class);
         expectConfigValidation(connectorMock, true, connConfig);
 
         // Should get first config
-        connectorConfigCb.onCompletion(null, connConfig);
-        EasyMock.expectLastCall();
+        doNothing().when(connectorConfigCb).onCompletion(null, connConfig);
         // Update config, which requires stopping and restarting
-        worker.stopAndAwaitConnector(CONNECTOR_NAME);
-        EasyMock.expectLastCall();
-        Capture<Map<String, String>> capturedConfig = EasyMock.newCapture();
-        Capture<Callback<TargetState>> onStart = EasyMock.newCapture();
-        worker.startConnector(eq(CONNECTOR_NAME), EasyMock.capture(capturedConfig), EasyMock.anyObject(),
-                eq(herder), eq(TargetState.STARTED), EasyMock.capture(onStart));
-        EasyMock.expectLastCall().andAnswer(() -> {
+        doNothing().when(worker).stopAndAwaitConnector(CONNECTOR_NAME);
+        final ArgumentCaptor<Map<String, String>> capturedConfig = ArgumentCaptor.forClass(Map.class);
+        final ArgumentCaptor<Callback<TargetState>> onStart = ArgumentCaptor.forClass(Callback.class);
+        doAnswer(invocation -> {
             onStart.getValue().onCompletion(null, TargetState.STARTED);
             return true;
-        });
+        }).when(worker).startConnector(eq(CONNECTOR_NAME), capturedConfig.capture(), any(),
+            eq(herder), eq(TargetState.STARTED), onStart.capture());
         // Generate same task config, which should result in no additional action to restart tasks
-        EasyMock.expect(worker.connectorTaskConfigs(CONNECTOR_NAME, new SourceConnectorConfig(plugins, newConnConfig, true)))
-                .andReturn(singletonList(taskConfig(SourceSink.SOURCE)));
+        when(worker.connectorTaskConfigs(CONNECTOR_NAME, new SourceConnectorConfig(plugins, newConnConfig, true)))
+            .thenReturn(singletonList(taskConfig(SourceSink.SOURCE)));
 
         expectConfigValidation(connectorMock, false, newConnConfig);
-        connectorConfigCb.onCompletion(null, newConnConfig);
-        EasyMock.expectLastCall();
-        EasyMock.expect(worker.getPlugins()).andReturn(plugins).anyTimes();
-
-        PowerMock.replayAll();
+        doNothing().when(connectorConfigCb).onCompletion(null, newConnConfig);
```

Comment:
Can we move this and the line above later, after the call to `herder.putConnectorConfig(CONNECTOR_NAME, connConfig, false, createCallback)` (line 763) and before the call to `herder.putConnectorConfig(CONNECTOR_NAME, newConnConfig, true, reconfigureCallback)` (line 770)?

Reply:
Done and pushed.